lib/cfn-nag/custom_rules/VpcHasFlowLogRule.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class VpcHasFlowLogRule < BaseRule
+  def rule_text
+    'VPC should have a flow log attached'
+  end
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W58'
+  end
+
+  def audit_impl(cfn_model)
+    violating_policies = cfn_model.resources_by_type('AWS::EC2::VPC')
+                                  .select do
```

Comment:
i'm pretty sure this only works in a coincidental way for the degenerate case.  i believe this will select every vpc if there are no flow logs..... and it will select no vpcs if there is at least one flow log

Reply:
to pick out individual vpcs without flow logs, you'll need to establish the relationship between the model elements and then query against that.

Reply:
@erickascic , I am definitely willing to do that if it's required, but I was going off the comments left by @twellspring on the ticket where he brought up that the flow logs might be attached to subnets rather than the VPC itself. I guess which is the lesser of the two evils?

Reply:
one way or the other, this impl has no traceability back to the attached resource.  as far as subnets go... we could consider them with by synthesizing the model a bit to look for attachments in both places.... on the other hand, i would consider _why_ would the attachment only be at the subnet level?  i have yet to see a security org that singles out particular subnets for compliance, and leaves others out.